components/modal/aria-helper.js

```diff
@@ -0,0 +1,77 @@
+/**
+ * External dependencies
+ */
+import { forEach } from 'lodash';
+
+let hiddenElements = [],
+	isHidden = false;
+
+/**
+ * Hides all elements in the body element from screen-readers except
+ * the provided element and elements that should not be hidden from
+ * screen-readers.
+ *
+ * The reason we do this is because `aria-modal="true"` currently is bugged
+ * in Safari, and support is spotty in other browsers overall. In the future
+ * we should consider removing these helper functions in favor of
+ * `aria-modal="true"`.
+ *
+ * @param {Element} unhiddenElement The element that should not be hidden.
+ */
+export function hideApp( unhiddenElement ) {
+	if ( isHidden ) {
+		return;
+	}
+	const elements = document.body.children;
+	forEach( elements, ( element ) => {
+		if (
+			element === unhiddenElement
+		) {
+			return;
+		}
+		if ( elementShouldBeHidden( element ) ) {
+			element.setAttribute( 'aria-hidden', 'true' );
+			hiddenElements.push( element );
+		}
+	} );
+	isHidden = true;
+}
+
+/**
+ * Determines if the passed element should not be hidden from screen readers.
+ *
+ * @param {HTMLElement} element The element that should be checked.
+ *
+ * @return {boolean} Whether the element should not be hidden from screen-readers.
+ */
+export function elementShouldBeHidden( element ) {
+	const liveRegionAriaRoles = [
+		'alert',
+		'status',
+		'log',
+		'marquee',
+		'timer',
+	];
+	const role = element.getAttribute( 'role' );
+	return ! (
+		element.tagName === 'SCRIPT' ||
+		element.hasAttribute( 'aria-hidden' ) ||
+		element.hasAttribute( 'aria-live' ) ||
+		liveRegionAriaRoles.includes( role )
```

Comment:
- `Array#includes` is not polyfilled and does not exist in IE11. This will throw errors.
- For constant sets of values, I might be more suggestive of using `Set` and [`Set#has`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set/has), though I'm not entirely certain of this yet. In theory, `Set` should be more optimized for this behavior.
  - Natively can have better performance characteristics than `Array` and `Array#indexOf`, depending on layers of polyfill abstraction and number of members ([jsperf](https://jsperf.com/array-indexof-vs-core-set-has)).
- The `liveRegionAriaRoles` array is a constant value and doesn't need to be redefined in the scope of the function. We're causing the garbage collector to do more work than necessary. This could be a top-of-file `LIVE_REGION_ARIA_ROLES` variable.